tier3/Dockerfile

```
@@ -30,6 +30,13 @@ RUN (cd /opt && \
     if [ "$(arch)" = x86_64 ]; then \
         mkdir /opt/lean && \
         curl --compressed -L "https://github.com/leanprover/lean4/releases/download/v4.0.0-m2/lean-4.0.0-m2-linux.tar.gz" | \
-        tar xz -C /opt/lean --strip-components=1; fi
+        tar xz -C /opt/lean --strip-components=1; fi && \
+    curl "https://packages.microsoft.com/config/debian/11/packages-microsoft-prod.deb" --output packages-microsoft-prod.deb && \
+    dpkg -i packages-microsoft-prod.deb && \
+    rm packages-microsoft-prod.deb && \
+    apt-get update; \
+    apt-get install -y apt-transport-https && \
```

```suggestion
    apt-get install -y apt-transport-https && \
```

I think this is already installed at this point.

---

`apt-transport-https` seems to be installed in the tier2 file [here](https://github.com/DMOJ/runtimes-docker/blob/master/tier2/Dockerfile#L5) already. I wasn't sure whether the different language installations should be "standalone" or whether it's safe to rely on previously installed dependencies.

---

Yep, it's safe because each tier is based off the previous one, e.g.: https://github.com/DMOJ/runtimes-docker/blob/33fe3adfa76d5523beb5fb8c3c756fc1f237b4df/tier3/Dockerfile#L1